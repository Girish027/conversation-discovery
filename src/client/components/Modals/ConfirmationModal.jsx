import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Dialog } from '@tfs/ui-components';
import constants from '../../constants';

export default class ConfirmationModal extends Component {
  constructor(props) {
    super(props);
    this.props = props;

    this.onClickOk = this.onClickOk.bind(this);
  }

  onClickOk() {
    const { onClickOk, onClickClose } = this.props;
    onClickClose();
    onClickOk();
  }

  render() {
    const {
      message, header, styleOverride, deleteMode, confirmMode, ...otherProps
    } = this.props;
    const { buttons } = constants;
    let { dontCancel: okChildren } = buttons;
    let { yesCancel: cancelChildren } = buttons;

    if (deleteMode) {
      okChildren = buttons.delete;
      cancelChildren = buttons.cancel;
    }

    if (confirmMode) {
      okChildren = buttons.ok;
      cancelChildren = buttons.cancel;
    }

    return (
      <div>
        <Dialog
          size='small'
          headerChildren={header}
          {...otherProps}
          isOpen
          okVisible
          okChildren={okChildren}
          cancelVisible
          cancelChildren={cancelChildren}
          closeIconVisible={false}
          styleOverride={styleOverride}
          onClickOk={this.onClickOk}
          centerContent={false}
        >
          <div>
            {message}
          </div>
        </Dialog>
      </div>
    );
  }
}

ConfirmationModal.propTypes = {
  message: PropTypes.string.isRequired,
  header: PropTypes.string.isRequired,
  closeIconVisible: PropTypes.bool,
  onClickOk: PropTypes.func,
  onClickClose: PropTypes.func,
  onClickCancel: PropTypes.func,
  styleOverride: PropTypes.object,
  deleteMode: PropTypes.bool,
  confirmMode: PropTypes.bool,
};

ConfirmationModal.defaultProps = {
  closeIconVisible: false,
  deleteMode: false,
  confirmMode: false,
  onClickOk: () => { },
  onClickClose: () => { },
  onClickCancel: () => { },
  styleOverride: {},
};
