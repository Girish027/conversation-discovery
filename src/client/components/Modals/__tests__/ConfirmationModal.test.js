import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ConfirmationModal from 'components/Modals/ConfirmationModal';

describe('<ConfirmationModal />', function () {
  let wrapper;


  const props = {
    header: 'Confirming action',
    message: 'Are you sure you want to do this?'
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (propsObj) => shallow(<ConfirmationModal
    {...propsObj}
  />);

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = getShallowWrapper(props);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly in confirmation mode', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly in delete confirm mode', () => {
      wrapper = getShallowWrapper({ ...props, deleteMode: true });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    beforeEach(() => {
      props.onClickClose = jest.fn();
      props.onClickOk = jest.fn();
      wrapper = getShallowWrapper(props);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    describe('onClickOk', () => {
      test('should close the dialog and call the okay handler', () => {
        wrapper.instance().onClickOk();
        expect(props.onClickClose).toHaveBeenCalled();
        expect(props.onClickOk).toHaveBeenCalled();
      });
    });
  });
});
