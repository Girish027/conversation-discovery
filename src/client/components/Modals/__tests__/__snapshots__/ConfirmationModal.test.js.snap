// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationModal /> Snapshots renders correctly in confirmation mode 1`] = `
<div>
  <withTheme(DialogComponent)
    cancelChildren="YES, CANCEL"
    cancelVisible={true}
    centerContent={false}
    closeIconVisible={false}
    headerChildren="Confirming action"
    isOpen={true}
    okChildren="NO, GO BACK"
    okVisible={true}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    size="small"
    styleOverride={Object {}}
  >
    <div>
      Are you sure you want to do this?
    </div>
  </withTheme(DialogComponent)>
</div>
`;

exports[`<ConfirmationModal /> Snapshots renders correctly in delete confirm mode 1`] = `
<div>
  <withTheme(DialogComponent)
    cancelChildren="CANCEL"
    cancelVisible={true}
    centerContent={false}
    closeIconVisible={false}
    headerChildren="Confirming action"
    isOpen={true}
    okChildren="DELETE"
    okVisible={true}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    size="small"
    styleOverride={Object {}}
  >
    <div>
      Are you sure you want to do this?
    </div>
  </withTheme(DialogComponent)>
</div>
`;
